devfiles/vscode-extensions.devfile.yaml

```diff
@@ -0,0 +1,79 @@
+# A devfile to develop VS Code extensions.
+# For developers who would like to create his plugin inside Che-Theia.
+
+apiVersion: 1.0.0
+metadata:
+  name: che-theia-all
+
+projects:
+
+  - name: che-theia
+    source:
+      type: git
+      location: 'https://github.com/eclipse/che-theia.git'
+
+components:
+
+  - id: eclipse/che-theia/next
+    type: cheEditor
+    alias: che-theia
+
+  - alias: che-dev
+    type: dockerimage
+    image: eclipse/che-theia-dev:next
+    mountSources: true
+    endpoints:
+      - name: "theia-dev-flow"
+        port: 3010
+        attributes:
+          protocol: http
+          public: 'true'
+    memoryLimit: "3Gi"
+
+  - id: redhat/vscode-yaml/latest
+    type: chePlugin
+
+  - id: che-incubator/typescript/latest
+    type: chePlugin
+    memoryLimit: 2048M
+
+commands:
+
+  - name: 1. generate ... Create extension
+    actions:
+      - workdir: /projects
+        type: exec
+        command: >
+          yes | yo code \
+            --extensionType="command-ts" \
+            --extensionName='helloworld-vscode' \
+            --extensionDescription='Helloworld extension for VS Code' \
+            --extensionDisplayName='helloworld-vscode' > /dev/null;
+          echo -e "\e[32mDone.\e[0m Generating complete"
+        component: che-dev
+
+  - name: 2. package ... Hello World extension
+    actions:
+      - workdir: /projects/helloworld-vscode
+        type: exec
+        command: >
+          npm install;
+          yes | vsce package;
+          echo -e "\e[32mDone.\e[0m Packaging complete"
+        component: che-dev
+
+  - name: 3. run ... HOSTED Che-Theia + Hello World extension
```

Comment:
what if the vscode extension needs to be tested in a sidecar container ? (like most of the vscode extension we have)

Reply:
I think it's possible. Let's create a dedicated PR with this improvement.

Reply:
this devfile is doing it: https://github.com/sunix/devfiles/blob/480188119d223d631d17472621e0cef618e22dc9/devfiles/vscode-quarkus/devfile.yaml

Reply:
I will create another PR with this command.
There is a big difference in use cases, that the workspace doesn't have a special sidecar container.
But I can add a command to launch Che-Theia in `theia-ide` container and to launch VS Code extension in `che-dev` container.